hack/setup-temporary-gopath.sh

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+
+# Conditionally create a temporary GOPATH for codegen
+# and openapigen to execute in. This is only done if
+# the current repo directory is not within GOPATH.
+function shim_gopath() {
+  local TEMP_GOPATH="/tmp/tekton-pipelines-codegen-gopath"
```

Comment:
Any reason to use than instead of `mktemp` ?
One reason to use `mktemp` would be that not all systems use `/tmp` as temporary directory, and usually `mktemp` handle this 🙃.

Reply:
I initially avoided that because `mktemp` on Mac has non-standard behaviour, meaning more shell code to work around it.  I don't feel strongly about it but in the moment writing it felt like just using `/tmp` was going to be clearer if / when it breaks on a system without a `/tmp` directory.